### ISSUE TITLE
Folding breaks when both 'begin' and 'end' markers are on the same line

### ISSUE THREAD

**Issue by HolyBlackCat** [zokugun/vscode-explicit-folding#4]
How to reproduce:

* Config:
```
"folding": {
    "*": [
        {
            "begin": "\\{\\{",
            "end": "\\}\\}",
        },
    ]
},
```

* File:
```
{{
    {{
        foo{{}}
    }}
}}
```

Here's how the result looks like:
![1](https://user-images.githubusercontent.com/20289055/47997476-02a51d80-e10d-11e8-99ed-af12b7726a81.png)

Now this might sound like an obscure thing to do, and it is, BUT...

Despite what the readme says, your extension seems to actually override/remove the builtin folding rules!  
It means that in addition to custom rules I have to manually readd the basic curly brace folding: `"begin":"{","end":"}"`.
I'm fine with that, but in the language I'm working with `{}` on a single line is a common thing, which breaks your extension. :(

**Comment by daiyam** [comment]
@HolyBlackCat do make sure you have `"editor.foldingStrategy": "auto"`.
Which language are you using?

The folding markers are mostly meant to be used in a comment line ([Emacs Folding Mode](https://www.emacswiki.org/emacs/FoldingMode))

Here what I get in a JavaScript file:
![screenshot](https://user-images.githubusercontent.com/587742/48001077-c9b67a00-e107-11e8-8e69-60fe0e5ee32f.jpg)

**Comment by HolyBlackCat** [comment]
Yes, `"editor.foldingStrategy"` is set to `"auto"`.

That's not a real language, just an example.

I get that behavior in a new unsaved file, without a language selected (status bar says `Plain Text`). The same problem persists in C++ files.

I'm not sure why it's different in JavaScript files, but even in those the third line of this snippet folds incorrectly:

![1](https://user-images.githubusercontent.com/20289055/48005159-c29d6500-e123-11e8-9eae-fb2500a4a286.png)

> markers are mostly meant to be used in a comment line

That was my original intention, but since the extension seems to prevent builtin folding rules for `.cpp` files from working (curly braces no longer fold), I needed to add `{"begin":"\\{","end":"\\}"}` rule as well, which breaks folding in a way similar to what I described above.

**Comment by daiyam** [comment]
@HolyBlackCat you are right there is an issue with c++ files.

**Comment by dfsbbl** [comment]
@daiyam Could you please try to fix this bug? Your extension is the only one I found in the VS Code markplace which can manually specify C/C++ folding marker, which is great for me, but this bug is a bit annoying when it occurs... Thank you!

**Comment by daiyam** [comment]
Hi @HolyBlackCat, it's very late but at last, it's fixed with 7ed6d0cc4e4d83b9a8a2a5b9ef43a9101a77bd29 (v0.4.0).
